src/main/kotlin/no/nb/mlt/wls/infrastructure/callbacks/InventoryNotifierAdapter.kt

```diff
@@ -1,21 +1,35 @@
 package no.nb.mlt.wls.infrastructure.callbacks
 
+import com.fasterxml.jackson.databind.ObjectMapper
 import no.nb.mlt.wls.domain.model.Item
 import no.nb.mlt.wls.domain.model.Order
 import no.nb.mlt.wls.domain.ports.outbound.InventoryNotifier
+import org.springframework.beans.factory.annotation.Value
 import org.springframework.stereotype.Component
 import org.springframework.web.reactive.function.client.WebClient
+import java.util.*
+import javax.crypto.Mac
+import javax.crypto.spec.SecretKeySpec
 
 @Component
 class InventoryNotifierAdapter(
-    private val webClient: WebClient
+    private val webClient: WebClient,
+    @Value("\${callback.secret}")
+    private val signatureSecretKey: String,
+    private val objectMapper: ObjectMapper
 ) : InventoryNotifier {
     override fun itemChanged(item: Item) {
+        val payload = objectMapper.writeValueAsString(item.toNotificationItemPayload())
+        val timestamp = System.currentTimeMillis().toString()
+        val signature = generateSignature(payload, timestamp)
+
         if (item.callbackUrl != null) {
             webClient
                 .post()
                 .uri(item.callbackUrl)
-                .bodyValue(item.toNotificationItemPayload())
+                .bodyValue(payload)
+                .header("X-Signature", signature)
+                .header("X-Timestamp", timestamp)
```

Comment:
When testing this locally, I had issues getting the request to parse correctly. This was a side-effect of the request not declaring its content type, which is something we should do:

```suggestion
                .header("X-Timestamp", timestamp)
                .header(HttpHeaders.CONTENT_TYPE, MediaType.APPLICATION_JSON_VALUE)
```